integration/query_frontend_test.go

```diff
@@ -137,6 +138,11 @@ func runQueryFrontendTest(t *testing.T, testMissingMetricName bool, setup queryF
 	require.NoError(t, queryFrontend.WaitSumMetrics(e2e.Equals(1), "cortex_memcache_client_servers"))
 	require.NoError(t, queryFrontend.WaitSumMetrics(e2e.Greater(0), "cortex_dns_lookups_total"))
 
+	// Confirm that the query-frontend is not ready b/c the querier is no connected
+	resp, err := http.Get("http://" + queryFrontend.HTTPEndpoint() + "/ready")
```

Comment:
The previous call `s.StartAndWaitReady(queryFrontend, distributor, ingester)` waits until the readiness probe succeed, so this test is not expected to work.